sentry.php

```diff
@@ -76,16 +78,16 @@ public function onPageNotFound()
         $uri          = $this->grav['uri'];
         $url          = $uri->url;
 
-        $vars         = array(
-            'url'     => $url,
-            'time'    => $time,
-            'referer' => isset($_SERVER['HTTP_REFERER']) ? $_SERVER['HTTP_REFERER'] : '',
-        );
+        // Just add context data and unique per url fingerprint before throwing Exception
+        Sentry\configureScope(function (Sentry\State\Scope $scope) use($url, $time) {
+            $scope->setExtra('url', $url);
+            $scope->setExtra('time', $time);
+            $scope->setExtra('referer', isset($_SERVER['HTTP_REFERER']) ? $_SERVER['HTTP_REFERER'] : '');
 
-        $this->client->captureMessage('Page not found: %s', array($url), array(
-            'extra' => $vars,
-            'fingerprint' => ['{{ default }}', $url]
-        ));
+            $scope->setFingerprint(['{{ default }}', $url]);
+        });
+
+        throw new \RuntimeException('Page not found: '. $url, 404);
```

Comment:
Why throw RuntimeException?
With disabled debug mode will be served white page instead 404 page.